### DISCUSSION TITLE
Ender 3 V3 KE - M600 ERROR

### DISCUSSION THREAD

**Discussion by weismantelf** [Guilouz/Creality-Helper-Script-Wiki#596] [Issues]
Hello,
I have installed the M600 extension on my Ender 3 V3 KE. 
The macro working so far very good. But when you cancel
the print from MainSail, Klipper will hangup in an endless loop
in pause mode. The only thing which helps then is to reboot 
the device.

When I am cancel the print, the following things happen:

- The toolhead moves to parking position
- The toolhead moves immediately  back to printing position
- On the web interface the pause button showing the wheel endless:

![image](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/58665051/4460ef5c-2b63-4318-b7c7-9159b4a99dd7)

When I remove the M600 extension, everything is working fine.

**Comment by Guilouz** [comment]
It strange because cancelling print call CANCEL_PRINT macro. It have nothing to see with M600 or PAUSE macro.

I don't have KE, so I can't try it.

**Comment by weismantelf** [comment]
Hello,
I found the issue. 
When the M600 extension is installed, the code_macro.cfg will be modified.
The script is commenting out the gcode_macro RESUME.

But the script is not commenting out the whole macro definition. There is a
rest of 7 lines remaining:

`
#    {% if printer.extruder.can_extrude|lower == 'true' %}
#      G91
#      G1 E{E} F2100
#    {% else %}
#      {action_respond_info("Extruder not hot enough")}
#    {% endif %}  
#    RESUME_BASE {get_params}
`

These remaining lines are now extending the gcode_macro Pause, 
which will then cause the problem that I have described. 

When the whole gcode_macro Resume is commented out, everything
is working as expected.

**Reply by Guilouz** [reply]
Thanks I have fixed it.

**Reply by HERVE-3D** [reply]
Hello Guilouz,
First a great thank you for what you do for the community.
Thanks to you I have rooted my elder 3 ke and everything is fine. The only thing is this famous M600. Can you please tell me which menu item will install M600n macro (is it “useful macros”?).
Thanks A LOT you will save me!
